tests/Http/ActionTest.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+namespace Tests\Http;
+
+use Tests\TestCase;
+use Rogue\Models\Action;
+use Rogue\Models\Campaign;
+
+class ActionTest extends TestCase
+{
+    /**
+     * Test that a POST request to /actions creates a new action.
+     *
+     * POST /actions
+     * @return void
+     */
+    public function testCreatingAnAction()
+    {
+        // Create a campaign that the action will belong to.
+        $campaign = factory(Campaign::class)->create();
+        $actionName = $this->faker->sentence;
+
+        // Create an action;
+        $this->actingAsAdmin()->postJson('actions', [
+            'name' => $actionName,
+            'campaign_id' => $campaign->id,
+            'post_type' => 'photo',
+            'reportback' => 1,
+            'civic_action' => 0,
+            'scholarship_entry' => 1,
+            'noun' => 'things',
+            'verb' => 'done',
+        ]);
+
+        // Make sure the action is persisted.
+        $this->assertDatabaseHas('actions', [
+            'name' => $actionName,
+        ]);
+    }
+
+    /**
+     * Test that a GET request to /api/v3/actions returns an index of all campaigns.
+     *
+     * GET /api/v3/actions
+     * @return void
+     */
+    public function testActionIndex()
+    {
+        // Create a campaign that the actions will belong to.
+        $campaign = factory(Campaign::class)->create();
+
+        // Create five actions.
+        factory(Action::class, 5)->create(['campaign_id' => $campaign->id]);
+
+        $response = $this->getJson('api/v3/actions');
+        $decodedResponse = $response->decodeResponseJson();
+
+        $response->assertStatus(200);
+        $this->assertEquals(5, $decodedResponse['meta']['pagination']['count']);
+    }
+
+    /**
+     * Test that a GET request to /api/v3/actions/:action_id returns the intended action.
+     *
+     * GET /api/v3/actions/:action_id
+     * @return void
+     */
+    public function testActionShow()
+    {
+        // Create a campaign that the action will belong to.
+        $campaign = factory(Campaign::class)->create();
+
+        // Create five actions.
+        factory(Action::class, 5)->create(['campaign_id' => $campaign->id]);
+
+        // Create 1 specific action to search for.
+        $secondCampaign = factory(Campaign::class)->create();
+
+        $action = factory(Action::class)->create(['campaign_id' => $secondCampaign->id]);
+
+        $response = $this->getJson('api/v3/actions/' . $action->id);
+        $decodedResponse = $response->decodeResponseJson();
+
+        $response->assertStatus(200);
+        $this->assertEquals($action->id, $decodedResponse['data']['id']);
+    }
+
+    /**
+     * Test that a PATCH request to /actions/:action_id updates an action.
+     *
+     * PATCH /actions/:action_id
+     * @return void
+     */
+    public function testUpdatingAnAction()
+    {
+        // Create a campaign that the action will belong to.
+        $campaign = factory(Campaign::class)->create();
```

Comment:
Do you need to create the campaign first? [In the action factory](https://github.com/DoSomething/rogue/blob/master/database/factories/ModelFactory.php#L137) I think it will still assign it a `campaign_id`

Reply:
@katiecrane great q! We do need to create a campaign bc I _think_ if we don't, then there are no campaigns in the database. So the foreign key restraint throws an error because there is no `id` in the `campaigns` table that matches with this action.

Reply:
super smart, @katiecrane ! Thanks for that - updated!